depth_first_traversal/depth_first_traversal.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+
+/* Program to perform depth first search over a graph with a specific start node.
+ */
+
+/* 
+ @ar - graph represeantation using adjency matrix
+ @visited - array to mark the visited nodes
+ @size - size of the array
+ */
+ 
+void depth_first_search(int ar[5][5], int *visited, int start, int size) {
+	
+	visited[start] = 1;
+	printf("node: %d\n", start);
+	
+	for ( int j=0; j<size; j++) {
+		if(ar[start][j] == 1 && visited[j]==0) {
+			
```

Comment:
Avoid giving extra empty lines. Only give empty lines where necessary. (e.g. Adding a blank line helps to symbolize different steps of the code)